flake.nix

```diff
@@ -56,54 +84,19 @@
           #  - run integ tests
           #  - do common development operations (e.g. lint, debug, and manage repos)
           inherit system;
+          buildInputs = libcrypto_packages ++ common_packages;
+          # This env var can be over-ridden instead of recreating the shellHook.
+          S2N_LIBCRYPTO = "openssl-1.1.1";
           shellHook = ''
-            echo Setting up enviornment from flake.nix...
             export S2N_LIBCRYPTO=openssl-1.1.1
+            echo Setting up $S2N_LIBCRYPTO enviornment from flake.nix...
             export PATH=${openssl_1_1_1}/bin:${gnutls-3-7}/bin:$PATH
```

Comment:
this is still hard coded to 1.1.1. Maybe this should be derived from the `S2N_LIBCRYPTO` variable?

Reply:
Correct, this breaks with openssl3, and presumably others. Once https://github.com/aws/s2n-tls/issues/3963 is addressed- we should revisit this.